readthedocs/core/signals.py

```diff
@@ -62,4 +67,23 @@ def decide_if_cors(sender, request, **kwargs):  # pylint: disable=unused-argumen
 
     return False
 
+
+@receiver(pre_delete, sender=User)
+def delete_projects_and_organizations(instance, *args, **kwargs):
+    # Here we count the owner list from the projects that the user own
+    # Then exclude the projects where there are more than one owner
+    projects_id = (instance.projects.all().annotate(num_users=Count('users'))
+                                          .exclude(num_users__gt=1)
+                                          .values_list('id', flat=True))
+
+    # Here we count the users list from the organization that the user belong
+    # Then exclude the organizations where there are more than one user
+    oauth_organizations_id = (instance.oauth_organizations.annotate(num_users=Count('users'))
+                                                          .exclude(num_users__gt=1)
+                                                          .values_list('id', flat=True))
```

Comment:
OAuth organizations aren't currently shared, so `User.oauth_organizations.delete()` should be fine. We'll need tests here to ensure this is the case.

Reply:
I dont know whats the case actually. maybe it was m2m to give a organization user same permission?